analytic_entries_report_fiscalyear/models/analytic_entries_report.py

```diff
@@ -0,0 +1,123 @@
+        the sql view"""
+        # added joins to account_period & account_move_line
+        # added appropriate fields to select list and group by
+        super(AnalyticEntriesReport, self).init(cr)
+        cr.execute("select pg_get_viewdef(%s::regclass)", (self._table,))
+        for statement in sqlparse.parse(cr.fetchone()[0]):
+            current_keyword = None
+            for token in statement:
+                if token.is_keyword:
+                    current_keyword = token
+                if isinstance(token, sqlparse.sql.IdentifierList) and\
+                   current_keyword.value == 'SELECT':
+                    last = None
+                    for last in token:
+                        pass
+                    token.insert_after(last, sqlparse.sql.Token(
+                        sqlparse.tokens.Generic,
+                        ',coalesce(ml.period_id, p.id) as period_id,'
+                        'coalesce(p_from_move.fiscalyear_id, p.fiscalyear_id) '
+                        'as fiscalyear_id'
+                    ))
+                if isinstance(
+                    token,
+                    (sqlparse.sql.IdentifierList, sqlparse.sql.Parenthesis)
+                ) and current_keyword.value == 'FROM':
+                    def find_table(token):
+                        if isinstance(token, sqlparse.sql.Identifier) and\
+                           token.get_real_name() == 'account_analytic_line':
+                            return token
+                        if not isinstance(token, collections.Iterable):
+                            return
+                        for child_token in token:
+                            result = find_table(child_token)
+                            if result:
+                                return result
+
+                    table = find_table(token)
+                    assert table
+
+                    table.parent.insert_after(table, sqlparse.sql.Token(
+                        sqlparse.tokens.Generic,
+                        ' left outer join account_period p '
+                        'on p.special = False and p.date_start <= a.date '
+                        'and p.date_stop >= a.date '
+                        'left outer join account_move_line ml '
+                        'on a.move_id = ml.id '
+                        'left outer join account_period p_from_move '
+                        'on ml.period_id = p_from_move.id '))
+                if isinstance(token, sqlparse.sql.IdentifierList) and\
+                   current_keyword.value == 'BY':
+                    last = None
+                    for last in token:
+                        pass
+                    token.insert_after(last, sqlparse.sql.Token(
+                        sqlparse.tokens.Generic,
+                        ', coalesce(p_from_move.fiscalyear_id,'
+                        'p.fiscalyear_id),'
+                        'coalesce(ml.period_id, p.id)'))
```

Comment:
I don't know why this suddenly pops up in the CI, but am getting:

```
File "/home/odorboprd11/odoo/parts/runbot/runbot/static/build/09255-8-0-merge-testing-89ae06/parts/odoo/openerp/addons/analytic_entries_report_fiscalyear/models/analytic_entries_report.py", line 79, in init
    (AsIs(self._table), AsIs(str(statement)[:-1])))
(...)
"ProgrammingError: column "ml.period_id" must appear in the GROUP BY clause or be used in an aggregate function
LINE 16:     sum(a.unit_amount) AS unit_amount,coalesce(ml.period_id,..."
```

Reply:
can you inspect the statement generated? my guess would be this code mixes up a group by with an order by clause

Reply:
this is an update in sqlparse: https://sqlparse.readthedocs.io/en/latest/changes/#release-0-3-0-mar-11-2019

`GROUP BY` is now parsed as one keyword, I updated the branch accordingly